x-pack/plugins/files/public/components/upload_file/util/parse_file_name.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import mime from 'mime-types';
+
+interface Result {
+  name: string;
+  mime?: string;
+}
+
+export function parseFileName(fileName: string): Result {
+  return {
+    name: fileName
+      .slice(0, 256)
+      .trim()
+      .replace(/\..*$/, '') // remove extension
```

Comment:
I think it won't be good to take out everything after the first dot. What about something that doesn't use a regex do remove the extension, like:
```js
  let parsed = fileName.slice(0, 256).trim();

  // remove the extension
  const lastDot = parsed.lastIndexOf('.');
  if (lastDot > 0) {
    parsed = parsed.substring(0, lastDot);
  }

  // replace invalid chars
  parsed = parsed.replace(/[^a-z0-9\s]/gi, '_');

  return {
    name: parsed,
    mime: mime.lookup(fileName) || undefined,
  };
```

Reply:
It looks like this is an old commit (I branched of this branch and needed to cherry-pick the change 🤦🏻 ). Updated!